### PR TITLE
feat(iOS): Implement blurEffect for new architecture

### PR DESCRIPTION
**Pull request by janicduplessis**
## Description

Implements blurEffect header config for new arch.

## Changes

 I kept this super simple and used string for codegen so we can simply reuse the RCTConvert implementation.

## Test code and steps to reproduce

Tested that header blur now works in an app with new arch enabled.

## Checklist

- [x] Included code example that can be used to test this change
- [x] Updated TS types
- [x] Updated documentation: 
  - [x] https://github.com/software-mansion/react-native-screens/blob/main/guides/GUIDE_FOR_LIBRARY_AUTHORS.md
  - [x] https://github.com/software-mansion/react-native-screens/blob/main/native-stack/README.md
  - [x] https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx
  - [x] https://github.com/software-mansion/react-native-screens/blob/main/src/native-stack/types.tsx
- [x] Ensured that CI passes

### CODE REVIEW

File: src/fabric/ScreenStackHeaderConfigNativeComponent.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export interface NativeProps extends ViewProps {
   backButtonDisplayMode?: WithDefault<BackButtonDisplayMode, 'default'>;
   hideBackButton?: boolean;
   backButtonInCustomView?: boolean;
+  blurEffect?: string;
```
Comment (tboba) [inline src/fabric/ScreenStackHeaderConfigNativeComponent.ts:47]: I don't know if I'm fond of setting blurEffect as a string. In our `types.tsx`, we're using a special [type](https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx#L42) for defining every possible string that could be a blurEffect. I'm concerned if there could be a situation, where invalid `blurEffect` entered here could just bring a crash on iOS. Some day ago, there was [that](https://github.com/software-mansion/react-native-screens/pull/1751) PR that was also adding a support for `blurEffect` on Fabric. Sadly, it has been abandoned, but maybe you could try to actually enter the type to the Spec and check if it'll just work the same as with `string`?

**Review by tboba — CHANGES_REQUESTED** [review]
Hi, thanks for creating this PR! I left a comment about the way how we're defining that prop in a Spec.

**Comment by janicduplessis** [comment]
@tboba Thanks for having a look. I will verify how incorrect values are handled. I liked the simplicity of this approach since it can use the same conversion code as the old arch, but I can also implement it using an enum, it will just require an extra function to convert from the codegen enum to the uikit one.

**Comment by janicduplessis** [comment]
Looks like there is also an android build issue I will need to fix.

**Comment by janicduplessis** [comment]
@tboba Update to use enum and fixed android build!

**Comment by janicduplessis** [comment]
@tboba Friendly ping on this :)

**Comment by tboba** [comment]
Hey hey @janicduplessis, yeah I'll try to check that today 👍 Thanks for the nudge :D

**Review by tboba — APPROVED** [review]
Everything looks good to me! 🫡 
Let's proceed when builds will pass 👍

**Pull request merged**